cartography/data/jobs/cleanup/aws_import_principals_cleanup.json

```diff
@@ -0,0 +1,8 @@
+{
+  "statements": [{
+    "query": "MATCH (:AWSAccount{id: {AWS_ID}})-[:RESOURCE]->(n:AWSPrincipal) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```

Comment:
In other places we have a second cleanup statement for the relationship like:
```
"MATCH (:AWSAccount{id: {AWS_ID}})-[r:RESOURCE]->(:AWSPrincipal) WHERE r.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted
```

This would handle the case where the nodes are up-to-date but the relationship between them no longer exists (although I'm not sure if this happens in this specific scenario, like a root Principal can't exactly change what account it's associated with right?).

Still, it would be good to explicitly delete stale relationships since that's a pattern we use in a bunch of other places.

Reply:
We don't follow that pattern for RESOURCE relationships due to the reason you described. Instead we rely on DETACH DELETE to do the right thing and remove the RESOURCE relationship when cleaning up stale nodes.

Reply:
Fair point 👍